nep-22.mediawiki

```diff
@@ -0,0 +1,71 @@
+<pre>
+  NEP: 22
+  Title: Contract Update Function
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>, Fernando Díaz Toledano <shargon@gmail.com>, Erik Zhang <erik@neo.org>, Roman Khimov <roman@nspcc.ru>
+  Type: Standard
+  Status: Accepted
+  Created: 2021-04-19
+</pre>
+
+
+==Abstract==
+
+This proposal outlines a standard for the NEO blockchain that will provide systems with a generalized interaction mechanism for smart contract update.
+
+==Motivation==
+
+As the Neo blockchain scales, Smart Contract Update will become increasingly important. We need a standard for <code>update</code> method in contract to callback native <code>ContractManagement</code> correctly.
+
+==Specification==
+
+All contracts SHOULD implement this NEP in case of any bug hotfix or improvement. If no update method there, incorrect action can never be corrected.
```

Comment:
Maybe `... implement this NEP to be able to fix bugs or otherwise improve functionality`. Just to not be misinterpreted for "implement after you have a bug", no, it's "think about it before you have bugs".